### PR TITLE
[iOS/Catalyst] Fix Shell TitleView rendering issues on iOS 16 

### LINKED ISSUE #10128: Shell.TitleView does not render on iOS 16
### Description

.NET MAUI on iOS 16 does not render a titleview. Versions 15.4 and below do render a title view

First noticed this in forms but it appears to be happening in MAUI as well
https://github.com/xamarin/Xamarin.Forms/issues/15542

### Steps to Reproduce

1. File -> New Maui App
2. Create a TitleView in any page
3. It won't render on an iOS 16 device.

Screenshot from Android Emulator (WORKING):
image

Screenshot from iPhone 12 iOS 16 (NOT WORKING):
84feca32-009a-4343-b131-a01fcd8fbb0d

### Link to public reproduction project repository

https://github.com/edgiardina/Maui-iOS16-TitleViewBug

### Version with bug

6.0.486 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS 16

### Did you find any workaround?

Have not figured out a workaround.

### Relevant log output

_No response_

### LINKED ISSUE #11309: [iOS] Shell.TitleView doesn't show accurate if not using just a string binding
### Description

When setting the `Shell.TitleView` in XAML, the content isn't shown up accurate on iOS 16.1.

```xaml
<Shell.TitleView>
        <Grid
            ColumnDefinitions="*,Auto"
            BackgroundColor="Red"
            >
            <Label
                Text="Headline"
                />
            <Button
                Grid.Column="1"
                Text="Button"
                />
        </Grid>
    </Shell.TitleView>
```
image

The `Label` isn't shown at all, the `Button` is only displayed half and setting a `BackgroundColor` has no effect (just for seeing the bounds of the `Grid`.
I just updated my repro for another issue with this issue.

Android works fine with the same project.

image

### Steps to Reproduce

1. Create a new MAUI app
2. Add the snippet above to the `MainPage`
3. Run it on iOS

### Link to public reproduction project repository

https://github.com/AndreasReitberger/ReproScrollingIssueModal

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS 16.1

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change

Fix Shell TitleView rendering issues on iOS 16.

shell-titleview

To validate and test the changes launch the Shell sample in the Gallery and validate that:
- The TitleView renders correctly.
- The second test binded to the Context is showing a Test.
- The Clicked event from the TitleView Button works (change the Tabbar background color).

Issue 11309:
shell-titleview-11309

Device Test:
shell-titleview-test

### Issues Fixed

Fixes #10128
Fixes #11309

### CODE REVIEW

**Review by jfversluis — APPROVED** [review]

**Review by mandel-macaque — APPROVED** [review]

File: src/Controls/src/Core/Compatibility/Handlers/Shell/iOS/ShellPageRendererTracker.cs [hunk 1]
```diff
@@ -551,7 +551,7 @@ public override void WillMoveToSuperview(UIView newSuper)
 
 			void UpdateFrame(UIView newSuper)
 			{
-				if (newSuper != null)
+				if (newSuper != null && newSuper.Bounds != CGRect.Empty)
```
Comment (mandel-macaque) [inline src/Controls/src/Core/Compatibility/Handlers/Shell/iOS/ShellPageRendererTracker.cs:554]:
```suggestion
				if (newSuper is not null && newSuper.Bounds != CGRect.Empty)
```

**Review by mandel-macaque — APPROVED** [review]
minor thing missing.

**Review by rmarinho — APPROVED** [review]

**Comment by FM1973** [comment]
When can we expect a backport to .net 7?

**Pull request merged**